### PR TITLE
feat(useAsyncQueue): new function

### LINKED ISSUE #899: feat: useAsyncQueue
I have an idea:

Provide an  controller that executes each asynchronous task sequentially .

```js
const { activeIndex, results } = useAsyncQueue([p1, p2, p3], {
  onError: () => {},
  onFinished: () => {},
})
```

### PR DESCRIPTION
**Pull request by webfansplz**
close #899

### CODE REVIEW

File: packages/core/useAsyncQueue/index.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import { noop } from '@vueuse/shared'
+
+interface Result {
+  state: 'pending' | 'fulfilled' | 'rejected'
+  data: any
+}
+
+type Task = (...args: any[]) => Promise<unknown>
+
+export interface useAsyncQueueOptions {
```
Comment (antfu) [inline packages/core/useAsyncQueue/index.ts:11]:
```suggestion
export interface UseAsyncQueueOptions {
```

File: packages/core/useAsyncQueue/index.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { ref, Ref } from 'vue-demi'
+import { noop } from '@vueuse/shared'
+
+interface Result {
+  state: 'pending' | 'fulfilled' | 'rejected'
+  data: any
+}
```
Comment (antfu) [inline packages/core/useAsyncQueue/index.ts:7]:
```suggestion
interface UseAsyncQueueResult<T> {
  state: 'pending' | 'fulfilled' | 'rejected'
  data: T
}
```

We'd better support generic types

**Review by antfu — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@vueuse/monorepo",
-  "version": "6.8.0",
+  "version": "6.7.6",
```
Comment (antfu) [inline package.json:3]: 👀

Comment (webfansplz) [reply package.json]: Sry,sending a lot of conflicts, I will close this PR and resubmit 😂

**Pull request closed without merging**